### ISSUE TITLE
When an empty array is in the jsonstring, jsonPart[0] fails.

### ISSUE THREAD

**Issue by Zhoot** [AngelQuirogaM/NiceJson#2]
When a empty array is in the json string, this part of the parse will fail.

(Line 372 - 376) of NiceJson.json

```
private static JsonNode ParseJsonPart(string jsonPart)
{
    JsonNode jsonPartValue = null;
    switch (jsonPart[0])
    {
```

**Comment by AngelQuirogaM** [comment]
I have been testing it , but no exceptions has happened at least with this test:

string jsonEmptyCurly = "{\"testing\":[]}";
JsonNode.ParseJsonString(jsonEmptyCurly);

string jsonEmptyCurly = "[]";
JsonNode.ParseJsonString(jsonEmptyCurly);

Please ¿Could you provide an example when it fails?

**Comment by Zhoot** [comment]
![image](https://cloud.githubusercontent.com/assets/4124559/18255747/e8917288-73ac-11e6-9fed-4868dc95195c.png)

Here's the entire json that's being used.
https://gist.github.com/Zhoot/041bf8c4b21b5a88722a720eefdd3c59#file-save-json-L23

**Comment by AngelQuirogaM** [comment]
Downloaded, tested and printed out using Unity3D. All seems to be working OK.

By your error number it could be that you have an old version of NiceJson.cs . Could you download it again and/or execute test scene in the test project?

If you are not using Unity3D in order to figure it out what is happening please tell me, What is your development platform?

**Comment by Zhoot** [comment]
Hey
Sorry was a bit busy.
I downloaded the latest version (Was on 1.2 when I found the errors) and I guess you fixed it in 1.3. Either on purpose or not 👍  Nice tool mate. I love it.
